### DISCUSSION TITLE
named-refused

### DISCUSSION THREAD

**Discussion by nixnutnz** [fail2ban/fail2ban#3838] [General]
could someone tell me what the failregex should be for this log entry please? have tried everything i've seen on the net to no avail
```
19-Sep-2024 18:18:39.454 query-errors: info: client @0x7f4f2c001b68 157.230.97.126#34060 (ta6.ch): query failed (REFUSED) for ta6.ch/IN/A at ../../../lib/ns/query.c:5486
```

**Comment by sebres** [comment]
If you have no filter yet (new jail):
```ini
failregex = ^\s*query-errors: info: client (?:@\S* )?<ADDR>#\S+(?: \([\S.]+\))?: query failed \(REFUSED\)
```
If you're thinking to extend `named-refused` filter (or adjust jail), one have to update `prefregex` additionally (let alone the format looks a bit different).
Something like that would match (one doesn't not even to specify additional `failregex` - adjusting of `prefregex` would enough):
```ini
[named-refused]
prefregex = ^<__line_prefix>(?:(?:error|info):\s*)?client(?: @\S*)? <ADDR>#\S+(?: \([\S.]+\))?: <F-CONTENT>.+</F-CONTENT>\s(?:(?:denied|failed)(?: \([^\)]*\))?|\(NOTAUTH\))(?: for \S+)?(?: at \S+)?\s*$
enabled = true
```
But one'd test this with `fail2ban-regex` against current log-file or journal, to avoid possible unwanted matches or even false positives.

**Reply by nixnutnz** [reply]
thank you sergey, that works well
there already several ip's banned but i have 1 ip that has been added yet keeps getting in, fail2ban says it's already banned which it is but still iptables is allowing it in, weird, i will search for an answer

**Reply by sebres** [reply]
See the answer for 2nd Q in [wiki :: FAQ](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works)...

Shortly, I guess it may be UDP, but banned was TCP only... The `named` jail example in default config:
https://github.com/fail2ban/fail2ban/blob/363c0d5fd074abfb394acdea91d547727881ac17/config/jail.conf#L723-L748

Or see it how it is made for nsd jail:
https://github.com/fail2ban/fail2ban/blob/363c0d5fd074abfb394acdea91d547727881ac17/config/jail.conf#L752-L753

But with newer versions one can also combine it to single jail (if banning action supports multiple protocols like `tcp,udp` or `all`, e. g. like current `iptables`-based actions):
```ini
[named]
filter = ...
...
protocol = tcp,udp
port = domain,953
banaction = iptables-ipset
enabled = true
```
